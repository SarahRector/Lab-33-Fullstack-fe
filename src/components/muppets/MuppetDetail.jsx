import React from 'react';
import PropTypes from 'prop-types';

const MuppetDetail = ({ name, performer, image }) => (
  <div data-testid="details">
    <h1>{name}</h1>
    <p>Performed By: {performer}</p>
    <img src={image} alt={name} />
  </div>
);

MuppetDetail.propTypes = {
  name: PropTypes.string,
  performer: PropTypes.string,
  image: PropTypes.string
};

export default MuppetDetail;
