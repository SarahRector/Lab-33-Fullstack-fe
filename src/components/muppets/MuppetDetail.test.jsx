import React from 'react';
import { render, cleanup } from '@testing-library/react';
import MuppetDetail from './MuppetDetail';

describe('MuppetDetail component', () => {
  afterEach(() => cleanup());
  it('renders MuppetDetail', () => {
    const { asFragment } = render(<MuppetDetail
      name="Kermit"
      performer="Jim Henson"
      image="test.png" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
