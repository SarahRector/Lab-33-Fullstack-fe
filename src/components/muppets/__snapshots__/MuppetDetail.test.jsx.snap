// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MuppetDetail component renders MuppetDetail 1`] = `
<DocumentFragment>
  <div
    data-testid="details"
  >
    <h1>
      Kermit
    </h1>
    <p>
      Performed By: Jim Henson
    </p>
    <img
      alt="Kermit"
      src="test.png"
    />
  </div>
</DocumentFragment>
`;
